### PR TITLE
Update Hub to use Observables

### PR DESCRIPTION
**Pull request by elorzafe**
#### Description of changes

Using Observables instead of callbacks on `Hub`

#### Checklist

- [x] PR description included
- [ x `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by AllanZhengYP — APPROVED** [review]

File: packages/core/src/Hub.ts [hunk 1]
```diff
@@ -140,90 +100,27 @@ export class HubClass {
...
-				pattern: channel,
-				callback: cb,
-			});
-		} else {
-			let holder = this.listeners[channel];
-
-			if (!holder) {
-				holder = [];
-				this.listeners[channel] = holder;
+	listen(channel: string): Observable<HubCapsule> {
```
Comment (kvramyasri7) [inline packages/core/src/Hub.ts:106]: One question: Are we not supporting `RegExp` for `channel` anymore?🤔?

Comment (elorzafe) [reply packages/core/src/Hub.ts:106]: Good question, I think the regex is more useful when doing `filter`

For example: 

`Hub.listen('auth').pipe(filter(({ payload }) => payload.match(/user/))).subscribe(...)`

File: packages/core/src/Hub.ts
Comment (kvramyasri7) [inline packages/core/src/Hub.ts:1]: nit : can we remove `IPattern`, `IListener` intefaces if we are not using elsewhere

**Review by stocaaro — COMMENTED** [review]
Little bit unsure on where we landed with this one. Are we moving to Observables or keeping callbacks in V6?

**Comment by stocaaro** [comment]
This needs to be recreated or redirected to `next/main`.

**Pull request closed without merging**